src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -1795,6 +1795,91 @@ def elastic_pool_list_capabilities(
 
     return editions
 
+###############################################
+#                sql instance-pool            #
+###############################################
+
+
+def instance_pool_list(
+        client,
+        resource_group_name=None):
+    '''
+    Lists servers in a resource group or subscription
+    '''
+
+    if resource_group_name:
+        # List all instance pools in the resource group
+        return client.list_by_resource_group(
+            resource_group_name=resource_group_name)
+
+    # List all instance pools in the subscription
+    return client.list()
+
+
+def instance_pool_create(
+        cmd,
+        client,
+        instance_pool_name,
+        resource_group_name,
+        sku=None,
+        **kwargs):
+    '''
+    Creates a new instance pool
+    '''
+
+    kwargs['sku'] = _find_instance_pool_sku_from_capabilities(
+        cmd.cli_ctx, kwargs['location'], sku)
+
+    return client.create_or_update(
```

Comment:
Please use sdk_no_wait() for long running operation supporting --no-wait. e.g. https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/sql/custom.py#L1818

Reply:
Included